### PR TITLE
Detailed error reporting

### PR DESCRIPTION
**Pull request by haritamar**
Summary of the changes:

- Add context to anonymous tracking of exceptions, which is specific to the exception.
- Implement it particularly for DbtCommandError with useful information about the invocation.
- Add a mechanism to report internal exceptions (exceptions that are not fatal) in the cli-end event.

### CODE REVIEW

File: elementary/tracking/anonymous_tracking.py [hunk 1]
```diff
@@ -109,18 +117,32 @@ def track_cli_end(
...
-            "exception_type": str(type(exc)),
             "module_name": module_name,
             "command": command,
         }
+        props.update(self._get_exception_properties(exc))
+
         self.send_event("cli-exception", properties=props)
 
+    def record_cli_internal_exception(self, exc: Exception):
+        if len(self.internal_exceptions) < self.INTERNAL_EXCEPTIONS_LIMIT:
```
Comment (IDoneShaveIt) [inline elementary/tracking/anonymous_tracking.py:136]: Maybe when we pass the limitation we need to add a prop that tells us that?
So we could know at least how many problems there were in the run?

**Review by IDoneShaveIt — COMMENTED** [review]
Looking good!
Left some small comments 🙂

File: elementary/clients/gcs/client.py [hunk 2]
```diff
@@ -51,8 +55,10 @@ def send_report(
                     else "index.html",
                 )
                 logger.info("Updated GCS bucket's website.")
-        except google.cloud.exceptions.GoogleCloudError:
+        except google.cloud.exceptions.GoogleCloudError as ex:
             logger.exception("Failed to upload report to GCS.")
+            if self.tracking:
+                self.tracking.record_cli_internal_exception(ex)
```
[the remark this answers is not in the document]
Comment (haritamar) [reply elementary/clients/gcs/client.py:61]: So actually in this case the only thing we'll save is the type of the exception (see `_get_exception_properties`), like we do today for in the `cli-exception` event.
I only report additional information for Elementary exceptions, based on the exception type.

So should be fine?
You're right that the error message / traceback might contain sensitive data so I'm not reporting them.

File: elementary/tracking/anonymous_tracking.py (hunk 1 above)
Comment (haritamar) [reply elementary/tracking/anonymous_tracking.py:136]: Makes sense, I'll add

File: elementary/clients/gcs/client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (IDoneShaveIt) [reply elementary/clients/gcs/client.py:61]: Perfect 🙂

**Review by IDoneShaveIt — APPROVED** [review]
LGTM!

**Pull request merged**